runtime/lua/vim/_meta.lua

```diff
@@ -593,7 +593,23 @@ local create_option_metatable = function(set_type)
     -- To set a value, instead use:
     --  opt[my_option] = value
     _set = function(self)
-      local value = convert_value_to_vim(self._name, self._info, self._value)
+      local value
+      if self._value ~= nil then
+        value = convert_value_to_vim(self._name, self._info, self._value)
+      else
+        if set_type == SET_TYPES.LOCAL then
+          -- Equivalent to :setlocal option< in Vimscript which sets the local
+          -- value of the option to its global value by copying the value.
+          local global_opt = make_option(self._name, a.nvim_get_option_value(self._name, {scope = "global"}))
+          value = global_opt._value
+        elseif set_type == SET_TYPES.SET then
+          -- Equivalent to :set option< in Vimscript, which removes the local value
+          -- of the option so that the global value will be used. This only
+          -- works for global-local options
+          return a.nvim_set_option_value(self._name, nil, {scope = "local"})
+        end
```

Comment:
can this section call `self:reset()` instead? or are they different.

We may also want to document that doing `vim.opt.option_name = nil` is the same as `:reset()` if that's the case.

Reply:
It's not quite the same: `reset()` resets the option to the default value (`:set option&` in Vim script) while this block handled the `:set option<` case.

`vim.opt.option_name = nil` is also equivalent to `:set option<` (so not the same as `:reset()`).